service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java

```diff
@@ -299,19 +299,15 @@ public LoanItem renewItem(String uin, String itemId) throws Exception {
     public Boolean getBlockStatus(String uin) throws Exception {
         JsonNode user = getUserByUin(uin);
 
-        if (Objects.nonNull(user) && user.isObject()) {
-            Boolean block = getAutomatedBlockStatus(getText(user, "/id"));
-            if (Objects.nonNull(block)) {
-                return block;
-            }
+        String userId = getText(user, "/id");
+        if (StringUtils.isNotEmpty(userId)) {
 
-            HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
-            throw new HttpClientErrorException(status,
-                String.format("%s: Unable to retrieve automated block status for user.", status.getReasonPhrase()));
+            return getAutomatedBlockStatus(userId) || getManualBlockStatus(userId);
         }
 
-        HttpStatus status = HttpStatus.NOT_FOUND;
-        throw new HttpClientErrorException(status, String.format("%s: Unable to find user.", status.getReasonPhrase()));
+        HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
+        throw new HttpClientErrorException(status,
```

Comment:
Is it the plan to return a 500 Internal Server Error instead of a 404 Not Found when the user by the uin is not found?

This implementation appears to return 500 when the user by the uin is not found, if I am reading this correctly.

I see that you converted the test for this to instead handle a 500.
The mocked response that is returned for that test is:
[src/test/resources/mock/response/patron-blocks/automatedBlocksEmpty.json](https://github.com/TAMULib/CatalogService/blob/4e49dae8c5ce12c978b10b6f64df158c40e65ee1/service/src/test/resources/mock/response/patron-blocks/automatedBlocksEmpty.json)

Reply:
The user has been found. It just did not have an id text node. The user not found exception is being thrown from the method `getUserByUin`. If the response body of that is malformed it is not really a not found, but rather a invalid response from FOLIO.